Creating Tests base and GetContainer test for RS Vault OneSDK

dragonfly91 opened the pull request:
Creating Tests base and GetContainer test for RS Vault OneSDK

Commit:
Creating Tests base and GetContainer test for OneSDK

sudhakar4171 commented on src/ResourceManagement/AzureBackup/RecoveryServices.Tests/ScenarioTests/RecoveryServicesTestBase.cs:
```diff
@@ -0,0 +1,114 @@
+{
+    public class RecoveryServicesTestsBase : TestBase
+    {
+        public static T GetServiceClient<T>(string resourceNamespace) where T : class
```
do you want to duplicate all these methods across backupvault and RSvault? How about creating some common which both projects reference

sudhakar4171 commented on src/ResourceManagement/AzureBackup/RecoveryServices.Tests/ScenarioTests/ContainerTests.cs:
```diff
@@ -0,0 +1,55 @@
+    public class ContainerTests : RecoveryServicesTestsBase
+    {
+        [Fact]
+        public void ListContainersTest()
```
Just think when MAB and DPM tests comes in future, do they need to duplicate all that code you write for IaaS? do we need to put some common logic in separate files

sudhakar4171 commented on src/ResourceManagement/AzureBackup/RecoveryServices.Tests/ScenarioTests/ContainerTests.cs:
```diff
@@ -0,0 +1,55 @@
+
+                    ProtectionContainerListQueryParams queryParams = new ProtectionContainerListQueryParams();
+                    queryParams.ProviderType = "AzureIaasVM";
+                    var response = client.Container.List("ASRRG1", "backupvault1", queryParams, GetCustomRequestHeaders());
```
do you want to create common constants file and define any hard-coded values

Commit:
1. Updating AzureBackup OneSDK with corresponding latest Hydra spec
2. Created Helpers folder which currently houses some Constants
3. Refactored test code to enable better reuse

dragonfly91 commented:
Updated the PR by incorporating the suggestions given by Sudreddy

sudhakar4171 commented on src/ResourceManagement/AzureBackup/RecoveryServices.Tests/ScenarioTests/AzureIaasVMTests/ContainerTests.cs:
```diff
@@ -0,0 +1,64 @@
+            ExecuteTest(
+                client =>
+                {
+                    string rsVaultRgName = GetSetting(TestConstants.RsVaultRgName);
```
If our intention is to use same vault for all tests, then it is better to move this common logic as well into containerhelper.cs.

Commit:
Created ContainerTestHelper class and moved some common code for ContainerTests

Commit:
Container Test Helper Changes
